src/amber/support/client_reload.cr

```diff
@@ -29,19 +34,38 @@ module Amber::Support
       end
     end
 
-    private def watch_files
-      watch(["public/**/*"]) do |event|
-        event.on_change do |files|
-          files.each do |file, timestamp|
-            Amber.logger.puts "Watching file: ./#{file}", "Watcher", :light_gray
-            case file
-            when .ends_with? ".css"
-              reload_clients(msg: "refreshcss")
-            else
-              reload_clients(msg: "reload")
-            end
-          end
+    private def check_file(file)
+      case file
+      when .ends_with? ".css"
+        reload_clients(msg: "refreshcss")
+      else
+        reload_clients(msg: "reload")
+      end
+    end
+
+    private def get_timestamp(file : String)
+      File.stat(file).mtime.to_s("%Y%m%d%H%M%S")
+    end
+
+    private def scan_files
+      file_changed = false
+
+      Dir.glob(["public/**/*"]) do |file|
+        timestamp = get_timestamp(file)
+
+        if FILE_TIMESTAMPS[file]? && FILE_TIMESTAMPS[file] != timestamp
+          FILE_TIMESTAMPS[file] = timestamp
+          file_changed = true
+          Amber.logger.puts "./#{file.colorize(:light_gray)}", "Watcher", :light_gray
+        elsif FILE_TIMESTAMPS[file]?.nil?
+          FILE_TIMESTAMPS[file] = timestamp
+          file_changed = true
+          Amber.logger.puts "Watching file: ./#{file.colorize(:light_gray)}", "Watcher", :light_gray
```

Comment:
Do the messages need to be different - one saying `Watching file: ...` the other, looks the same other than that prefix.

Could this be simplified?
```crystal
private def scan_files
  Dir.glob(["public/**/*"]) do |file|
    timestamp = get_timestamp(file)
    msg_prefix = FILE_TIMESTAMPS[file]? ? "" : "Watching file: "

    if FILE_TIMESTAMPS[file]? != timestamp
      FILE_TIMESTAMPS[file] = timestamp
      Amber.logger.puts "#{msg_prefix}./#{file.colorize(:light_gray)}", "Watcher", :light_gray
      check_file(file)
    end
  end
end
```

Reply:
Thank you! done 👍